nomt/benches/beatree.rs

```diff
@@ -1,7 +1,41 @@
-use criterion::{criterion_group, criterion_main, Criterion};
+#![cfg(feature = "benchmarks")]
 
-pub fn beatreee_benchmark(_c: &mut Criterion) {
-    todo!()
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use nomt::{separate, Key};
+use rand::RngCore;
+
+// returns two keys a and b where b > a and b shares the first n bits with a
+fn get_keys(shared_bytes: usize) -> (Key, Key) {
+    let mut rand = rand::thread_rng();
+    let mut a = [0; 32];
+    rand.fill_bytes(&mut a[0..shared_bytes]);
+
+    // b > a
+    let mut b = a.clone();
+    b[shared_bytes] = 1;
+
+    (a, b)
+}
+
+fn separate_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("separate");
+
+    for shared_bytes in 0..32 {
+        let input = get_keys(shared_bytes);
+        group.bench_with_input(
+            BenchmarkId::from_parameter(shared_bytes),
+            &input,
+            |b, &input| {
+                b.iter(|| separate(&input.0, &input.1));
+            },
+        );
+    }
+
+    group.finish();
+}
+
+pub fn beatreee_benchmark(c: &mut Criterion) {
+    separate_benchmark(c);
```

Comment:
It would be better to have multiple benchmarks, each for their own number of shared _bits_

Reply:
It is already set up like this: having a benchmarking group and using `bench_with_input` results in multiple benchmarks. Currently, there are shared bytes ranging from 0 to 32 (I will reduce the range to what you described, as they otherwise take too long)

or do you mean something different?

an example of `cargo bench --features benchmarks` is:

```shell
separate/0              time:   [85.400 ns 85.514 ns 85.683 ns]
                        change: [+42.966% +44.113% +45.865%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 24 outliers among 100 measurements (24.00%)
  8 (8.00%) low mild
  5 (5.00%) high mild
  11 (11.00%) high severe
separate/1              time:   [127.01 ns 127.16 ns 127.42 ns]
                        change: [-0.0945% -0.0070% +0.1108%] (p = 0.91 > 0.05)
                        No change in performance detected.
Found 9 outliers among 100 measurements (9.00%)
  1 (1.00%) high mild
  8 (8.00%) high severe
separate/2              time:   [167.66 ns 168.08 ns 168.73 ns]
                        change: [-5.0945% -4.2770% -3.5234%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  4 (4.00%) high mild
  5 (5.00%) high severe
```